tests/network_rules_update_test.go

```diff
@@ -0,0 +1,222 @@
+package tests
+
+import (
+	"encoding/json"
+	"github.com/0xsoniclabs/sonic/gossip/contract/driverauth100"
+	"github.com/0xsoniclabs/sonic/opera/contracts/driverauth"
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/stretchr/testify/require"
+	"math/big"
+	"testing"
+)
+
+func TestNetworkRule_Update_RulesChangeDuringEpochHasNoEffect(t *testing.T) {
```

Comment:
nit:
```suggestion
func TestNetworkRule_Update_RulesChangeIsDelayedUntilNextEpochStart(t *testing.T) {
```
Because the change has an effect, eventually.